stac_fastapi/opensearch/stac_fastapi/opensearch/database_logic.py

```diff
@@ -296,30 +284,18 @@ def apply_free_text_filter(search: Search, free_text_queries: Optional[List[str]
 
     @staticmethod
     def apply_datetime_filter(
-        search: Search, interval: Optional[Union[DateTimeType, str]]
+        search: Search, datetime_search: Dict[str, Optional[str]]
     ) -> Search:
         """Apply a filter to search on datetime, start_datetime, and end_datetime fields.
 
         Args:
             search: The search object to filter.
-            interval: Optional datetime interval to filter by. Can be:
-                - A single datetime string (e.g., "2023-01-01T12:00:00")
-                - A datetime range string (e.g., "2023-01-01/2023-12-31")
-                - A datetime object
-                - A tuple of (start_datetime, end_datetime)
+            datetime_search: Dict[str, Optional[str]]
 
         Returns:
             The filtered search object.
         """
-        if not interval:
-            return search
-
-        should = []
-        try:
-            datetime_search = return_date(interval)
-        except (ValueError, TypeError) as e:
-            # Handle invalid interval formats if return_date fails
-            logger.error(f"Invalid interval format: {interval}, error: {e}")
+        if not datetime_search:
```

Comment:
Maybe here we can re-implement the return_date function and remove it from core.